File: src/code-review.py
import os
import openai

def main():
    openai.api_key = os.environ['OPENAI_API_KEY']

    with open('diff.txt', 'r') as f:
        diff_content = f.read()

    # Truncate the diff if it's too long (adjust max_chars as needed)
    max_chars = 5000
    if len(diff_content) > max_chars:
        diff_content = diff_content[:max_chars]
        diff_content += '\n[Diff truncated due to length]'

    messages = [
        {"role": "user", "content": f"Please review the following code changes and provide constructive feedback:\n\n{diff_content}"}
    ]

    response = openai.ChatCompletion.create(
        model="gpt-4",
        messages=messages,
        temperature=0.2,
    )

    review = response.choices[0].message.content

    with open('review.txt', 'w') as f:
        f.write(review)

if __name__ == '__main__':
    main()

File: src/code-review.js
import fs from 'fs';
import { OpenAI } from 'openai';

const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

async function main() {
  let diffContent = fs.readFileSync('diff.txt', 'utf8');

  // Truncate the diff if it's too long (adjust maxTokens as needed)
  const maxTokens = 4000;
  const maxChars = maxTokens * 4;
  if (diffContent.length > maxChars) {
    diffContent = diffContent.substring(0, maxChars) + '\n[Diff truncated due to length]';
  }

  const messages = [
    {
      role: "user",
      content: `Please review the following code changes and provide constructive feedback:\n\n${diffContent}`
    }
  ];

  try {
    const response = await openai.chat.completions.create({
      model: "gpt-4",
      messages,
      temperature: 0.2,
    });

    const review = response.choices[0].message.content;
    fs.writeFileSync('review.txt', review);
  } catch (error) {
    console.error('Error communicating with OpenAI API:', error);
    fs.writeFileSync('review.txt', 'An error occurred while generating the code review.');
    process.exit(1);
  }
}

// Uncomment main to execute the script
main();
